src/main/java/com/splunk/ibm/mq/metricscollector/InquireChannelCmdCollector.java

```diff
@@ -64,40 +63,40 @@ public void publishMetrics() {
             "sending PCF agent request to query metrics for generic channel {}",
             channelGenericName);
         long startTime = System.currentTimeMillis();
-        PCFMessage[] response = context.send(request);
+        List<PCFMessage> response = context.send(request);
         long endTime = System.currentTimeMillis() - startTime;
         logger.debug(
             "PCF agent queue metrics query response for generic queue {} received in {} milliseconds",
             channelGenericName,
             endTime);
-        if (response == null || response.length <= 0) {
-          logger.warn("Unexpected Error while PCFMessage.send(), response is either null or empty");
+        if (response.isEmpty()) {
+          logger.warn("Unexpected error while PCFMessage.send(), response is empty");
           return;
         }
-        for (PCFMessage pcfMessage : response) {
-          String channelName =
-              pcfMessage.getStringParameterValue(CMQCFC.MQCACH_CHANNEL_NAME).trim();
-          Set<ExcludeFilters> excludeFilters = context.getChannelExcludeFilters();
-          if (!ExcludeFilters.isExcluded(
-              channelName, excludeFilters)) { // check for exclude filters
-            logger.debug("Pulling out metrics for channel name {}", channelName);
-            List<Metric> responseMetrics = Lists.newArrayList();
-            context.forEachMetric(
-                (metrickey, wmqOverride) -> {
-                  if (pcfMessage.getParameter(wmqOverride.getConstantValue()) == null) {
-                    logger.debug("Missing property {} on {}", metrickey, channelName);
-                    return;
-                  }
-                  int metricVal = pcfMessage.getIntParameterValue(wmqOverride.getConstantValue());
-                  Metric metric =
-                      metricCreator.createMetric(
-                          metrickey, metricVal, wmqOverride, channelName, metrickey);
-                  responseMetrics.add(metric);
-                });
-            context.transformAndPrintMetrics(responseMetrics);
-          } else {
-            logger.debug("Channel name {} is excluded.", channelName);
-          }
+
+        List<PCFMessage> messages =
+            MessageFilter.ofKind("channel")
+                .excluding(context.getChannelExcludeFilters())
+                .withResourceExtractor(MessageBuddy::channelName)
+                .filter(response);
+
+        for (PCFMessage message : messages) {
+          String channelName = MessageBuddy.channelName(message);
+          logger.debug("Pulling out metrics for channel name {}", channelName);
+          List<Metric> responseMetrics = Lists.newArrayList();
+          context.forEachMetric(
```

Comment:
sorry, same thing, IntelliJ is suggesting to extract this part into a method, so that it becomes... 

```
        for (PCFMessage message : messages) {
          String channelName = MessageBuddy.channelName(message);
          logger.debug("Pulling out metrics for channel name {}", channelName);
          List<Metric> responseMetrics = getMetrics(message, channelName);
          context.transformAndPrintMetrics(responseMetrics);
        }